handler/src/test/java/io/netty/handler/ssl/SslContextBuilderTest.java

```diff
@@ -326,6 +333,23 @@ private static void testServerContext(SslProvider provider) throws Exception {
         engine.closeOutbound();
     }
 
+    private static void testServerContextWithSecureRandom(SslProvider provider,
+                                                          SecureRandom mockSecureRandom) throws Exception {
+        SelfSignedCertificate cert = new SelfSignedCertificate();
+        SslContextBuilder builder = SslContextBuilder.forServer(cert.key(), cert.cert())
+                .sslProvider(provider)
+                .secureRandom(mockSecureRandom)
+                .trustManager(cert.cert())
+                .clientAuth(ClientAuth.REQUIRE);
+        SslContext context = builder.build();
+        SSLEngine engine = context.newEngine(UnpooledByteBufAllocator.DEFAULT);
+        assertFalse(engine.getWantClientAuth());
+        assertTrue(engine.getNeedClientAuth());
+        Mockito.verify(mockSecureRandom, Mockito.atLeastOnce()).nextInt();
```

Comment:
I'm not sure we can solely rely on `nextInt` being called. How about we have an inner class that extends `SecureRandom` and has a counter that's incremented on every method call?

Reply:
dce4e0149d7d261d5d953276e2c781718d9535a5